### PR TITLE
[SDKDG] Calls Exposed in the SDK

### PR DESCRIPTION
**Pull request by AliRKat**

### CODE REVIEW

File: sdk_dev_guide/current.md [hunk 1]
```diff
@@ -2541,6 +2541,47 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
   This way the event is sent as soon as possible to the server and marks the widget
   as "completed" for that specific user.
 </p>
+<h3 id="h_01HQMX59RNK27XW4S8HDMXBYC9">Calls Exposed in the SDK</h3>
+<p>
+  Feedback Widgets functionality requires specific calls to be exposed within SDK.
+  Although method names may vary, the core functionalities remain consistent. Here
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2547]: lets use a more normative language, like 'should'

File: sdk_dev_guide/current.md [hunk 2]
```diff
@@ -2541,6 +2541,47 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
...
+  are the primary feedback-related methods:
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2548]: mention that these methods should be exposed through an interface if applicable. that interface should be named 'feedback' etc

File: sdk_dev_guide/current.md [hunk 3]
```diff
@@ -2541,6 +2541,47 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
...
+</p>
+<ul>
+  <li>
+      <strong>Retrieve Available Feedback Widgets</strong>
+  </li>
+</ul>
+<pre><code>Countly.getAvailableFeedbackWidgets()</code></pre>
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2555]: examples should have an interface

File: sdk_dev_guide/current.md (hunk 1 above)
Comment (turtledreams) [reply sdk_dev_guide/current.md:2547]: also mention not all platforms will have all of these but this list is exhaustive

File: sdk_dev_guide/current.md [hunk 4]
```diff
@@ -2541,6 +2541,47 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
...
+<p>
+  Retrieves a list of available feedback widgets for the device ID.
+</p>
+<ul>
+  <li>
+      <strong>Present Feedback Widget</strong>
+  </li>
+</ul>
+<pre><code>Countly.presentFeedbackWidget()</code></pre>
+<p>Presents a chosen feedback widget in an alert dialog.</p>
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2565]: web shows it in iframe. alertdialog has a webview in it which actually shows the widget. so this sentence should be more generalizable without dissecting each option

File: sdk_dev_guide/current.md [hunk 5]
```diff
@@ -2541,6 +2541,47 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
...
+</p>
+<ul>
+  <li>
+      <strong>Report Feedback Widget Manually</strong>
+  </li>
+</ul>
+<pre><code>Countly.reportFeedbackWidgetManually()</code></pre>
+<p>
+  Manually reports a feedback widget in case a custom interface is used.
+</p>
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2584]: there is also a method that returns the url, no?

File: sdk_dev_guide/current.md [hunk 6]
```diff
@@ -2541,6 +2541,47 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
...
+<pre><code>Countly.presentFeedbackWidget()</code></pre>
+<p>Presents a chosen feedback widget in an alert dialog.</p>
+<ul>
+  <li>
+      <strong>Retrieve Feedback Widget Data</strong>
+  </li>
+</ul>
+<pre><code>Countly.getFeedbackWidgetData()</code></pre>
+<p>
+  Downloads data for a specific widget to be displayed with a custom UI. This data
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2573]: no need to explain what the returned data will be used for. you will explain that in its respective section

**Review by turtledreams — CHANGES_REQUESTED** [review]

**Review by turtledreams — APPROVED** [review]

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
needs to be tweaked as per slack info

File: sdk_dev_guide/current.md [hunk 7]
```diff
@@ -2541,6 +2541,42 @@ CountlyConfiguration.starRatingDismissButtonTitle = "Custom Dismiss Button Title
...
+  
+// Retrieves a specific widget's data
+Countly.feedback().getFeedbackWidgetData(CountlyFeedbackWidget widgetInfo, RetrieveFeedbackWidgetData callback {
+  public void onFinished(JSONObject retrievedWidgetData, String error) {
+    // Handling errors and retrieved widget's data, can be done here
+  }
+});
+  
+// Manually reports a widget's results
+Countly.feedback().reportFeedbackWidgetManually(CountlyFeedbackWidget widgetInfo, JSONObject widgetData, Map&lt;string, object&gt; widgetResult);
```
Comment (turtledreams) [inline sdk_dev_guide/current.md:2572]: mention these return sth or nothing(void)

**Review by turtledreams — CHANGES_REQUESTED** [review]

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
needs to be reworked

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
consent
no init time config options
where are the callback?
~ "the feedback interface should expost the following functions"

**Pull request merged**